### PR TITLE
Create authentication factory (angular) to process auth services.

### PR DESCRIPTION
**Pull request by tmartin1**
Add client side utility for the following operations with auth.service.js:
- login
- logout
- create user
- change password
- get current user
- check if user is logged in
- check is user is admin
- get token

Also adds PUT and GET definitions in user.service.js

### CODE REVIEW

**Comment by scottrice10** [comment]
1. Needs tests.
2. If lines of code are needed in the future, a TODO should be added in the comment. Sublime plugins will change the TODO colors, as well Webstorm, so it is easy to see what code needs to get implemented in the future.

File: client/components/auth/user.service.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+angular.module('badgerApp')
+  .factory('User', function ($resource) {
+    return $resource('/api/users/:id/:controller', {
+      id: '@_id'
```
Comment (scottrice10) [inline client/components/auth/user.service.js:7]: A TODO comment should be added here to specify that a controller parameter needs to be added in the future.

**Comment by tmartin1** [comment]
Clarified comments and added test file for auth components.

**Comment by scottrice10** [comment]
Looks like there is a merge conflict in client/index.html. Once that is resolved, this looks mergable.

**Pull request merged**